File: pdi/docs/hello_expose.c
#include <pdi.h>

int main(int argc, char* argv[]) {
    
    PDI_init(PC_parse_path("hello_data.yml"));
    int my_world = 42;
    
    //! [share_reclaim]
    PDI_share("world", &my_world, PDI_OUT);
    PDI_reclaim("world");
    //! [share_reclaim]
    
    //! [expose]
    PDI_expose("world", &my_world, PDI_OUT);
    //! [expose]

    PDI_finalize();
    return 0;
}

File: docs/First_steps/hello_expose.f90
program expose
    use paraconf
    use PDI
    implicit none

    type(PC_tree_t),target :: conf
    integer                :: my_world

    call PC_parse_path("hello_data.yml", conf)
    call PDI_init(conf)

    my_world = 42
    
    !! [share_reclaim]
    call PDI_share("world", my_world, PDI_OUT)
    call PDI_reclaim("world");
    !! [share_reclaim]
    
    !! [expose]
    call PDI_expose("world", my_world, PDI_OUT)
    !! [expose]

    call PDI_finalize()
end program expose
